test/tools/d_do_test.d

```diff
@@ -586,7 +586,7 @@ int tryMain(string[] args)
         switch (envData.os)
         {
             case "win32": envData.ccompiler = "dmc"; break;
-            case "win64": envData.ccompiler = `\"Program Files (x86)"\"Microsoft Visual Studio 10.0"\VC\bin\amd64\cl.exe`; break;
+            case "win64": envData.ccompiler = environment.get("VCBIN_DIR") ~ `\cl.exe`; break;
```

Comment:
Why was this done as a patch instead of pushing directly to the repo?

Anyway, we might want the logic to be a bit more "forgiving" and helpful in "bad environment" cases.  Maybe something like this:
```D
auto vcbinDir = environment.get("VCBIN_DIR", null);
if (!vcbinDir)
{
    // NOTE: I added "C:" because windows only installs Visual Studio to the C drive and without this
    //       you break 64-bit testing if dmd is on another drive.  I actually have dmd on another drive
    //       but I've never run into this problem since I've never gotten the dmd test suit to work on windows :)
    vcbinDir = `"C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\bin\amd64"`;
}
if (!exists(vcbinDir))
{
    writefln("Error: the Visual Studio C compiler bin directory does not exist (%s)", vcbinDir);
    return 1; // fail
}
envData.ccompiler = vcbinDir ~ `\cl.exe`;
```